support-frontend/assets/helpers/redux/subscriptionsStore.ts

```diff
@@ -0,0 +1,60 @@
+import { combineReducers, configureStore } from '@reduxjs/toolkit';
+import { renderError } from 'helpers/rendering/render';
+import { createWithDeliveryCheckoutReducer } from 'helpers/subscriptionsForms/subscriptionCheckoutReducer';
+import { setInitialCommonState } from './commonState/actions';
+import { commonReducer } from './commonState/reducer';
+import type { CommonState } from './commonState/state';
+import { initialCommonState } from './commonState/state';
+import { getInitialState } from './utils/setup';
+
+const subscriptionsPageReducer = createWithDeliveryCheckoutReducer(
+	initialCommonState.internationalisation.countryId,
+	'DigitalPack',
+	'Monthly',
+	null,
+	null,
+	null,
+);
+
+export type SubscriptionsReducer = typeof subscriptionsPageReducer;
+
+const baseReducer = {
+	common: commonReducer,
+	page: subscriptionsPageReducer,
+};
+
+const subscriptionsStore = configureStore({
+	reducer: baseReducer,
+});
+
+export type SubscriptionsStore = typeof subscriptionsStore;
+
+export function addPageReducer(
+	newReducer?: SubscriptionsReducer,
+): SubscriptionsStore {
+	return configureStore({
```

Comment:
Creating the store twice is obviously not ideal, but my initial approach using `store.replaceReducer` caused some odd problems when not in dev mode.

Reply:
Why is it that we need to do this?

Reply:
The subscriptions stores currently rely on creating a brand new page-specific reducer from scratch for each checkout, with specific initial state indicating the product, fulfilment option, etc etc, and we can't get around this without completely rewriting how all the checkout reducer code works. This means having no access to utility types because you don't know the structure of the store until runtime.

I initially took the approach of creating the static store instance and then using the [replaceReducer](https://redux.js.org/usage/code-splitting#using-replacereducer) utility to just update the store with the checkout-specific page reducer at runtime, but for god only knows what reason this did not work in browsers that do not have Redux devtools installed, hence tearing my hair out for half the day yesterday because the Selenium tests were failing with pages not even loading properly but I couldn't reproduce the failures! Creating a brand new store instance from scratch solves the problem.

In the future we can overhaul the store setup to not rely on creating reducers at runtime, but for the time being this means having access to the utility types without the checkouts breaking.

I should probably add or link to this explanation in a code comment actually, this is one of those 'I know this is a stupid decision but we have to' things that requires a bit of extra context.

Reply:
Ah I see - so creating the `subscriptionsPageReducer` on line 10 is just so that we can extract the utility types which we can then share across the checkout pages? And the utility types work across all those pages despite them having slightly different reducers?

Yeah I'd say it's worth a comment, as someone who knows much less about the checkout pages it was a little tricky to understand what was going on!

Reply:
Yes, exactly- the overall type structure of the checkout state is similar enough that the utility types should work in most/all cases, but it's the provision of the initial state for a given checkout with the relevant product details that's currently impossible to do without recreating the reducer from scratch.

I've added a link to this thread as a comment- should hopefully be helpful in future!